release/scan-image-vulnerabilities/scan-image-vulnerabilities.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+#
+# Scans an image for vulnerabilities and prints a summary of the vulnerabilities.
+# For the purpose of this script, a vulnerability is an image component in a version that is affected by a CVE.
+#
+# Local run:
+#
+#  test/local-env.sh release/scan-image-vulnerabilities/scan-image-vulnerabilities.sh <image> <summary-prefix>
+#
+set -euo pipefail
+
+function main() {
+  IMAGE="${1:-}"
+  SUMMARY_PREFIX="${2:-}"
+  local result_path="scan-result.json"
+
+  check_not_empty \
+      IMAGE \
+      SUMMARY_PREFIX
+
+  scan_image "$IMAGE" "$result_path"
+
+  # Count the number of vulnerabilities and fixable vulnerabilities for each severity.
+  # Use associative arrays to store counts by severity.
+  # Arrays are used via nameref parameters in print_vulnerability_status function.
+  # shellcheck disable=SC2034
+  declare -A vuln_counts
+  # shellcheck disable=SC2034
+  declare -A fixable_counts
+
+  # shellcheck disable=SC2034
+  for severity in CRITICAL IMPORTANT MODERATE LOW; do
+    vuln_counts[$severity]="$(count_vulnerabilities "$severity" "$result_path")"
+    fixable_counts[$severity]="$(count_fixable_vulnerabilities "$severity" "$result_path")"
+  done
+
+  gh_summary "### $SUMMARY_PREFIX $IMAGE"
+  print_vulnerability_status vuln_counts fixable_counts
+
+  # Print the vulnerabilities table in a collapsible section.
+  # For the table to render correctly, we need to add a newline after the summary.
+  gh_summary "<details><summary>Click to expand details</summary>\n"
+  gh_summary "$(print_vulnerabilities_table "$result_path")"
+  gh_summary "</details>"
+
+  # Fail the build if any fixable critical or important vulnerabilities are found.
+  severities=( "CRITICAL" "IMPORTANT" )
+  for severity in "${severities[@]}"; do
+    if (( fixable_counts[$severity] > 0 )); then
+      exit 1
+    fi
+  done
+}
+
+# Scans the image for vulnerabilities.
+function scan_image() {
+  local image="$1"
+  local result_path="$2"
+  roxctl image scan --output=json --force \
+    --image="${image}" | tee "$result_path"
```

Comment:
Bonus question: does `quay-bearer-token` get magically passed to `roxctl` or how does it know how to authenticate with quay.io?

Reply:
`roxctl` doesn't know, Central knows. In this case, ACS Prod Dogfooding (https://acs-cii18grublkv81uil8gg.acs.rhcloud.com/main/dashboard) knows. The README should make it clear that an image integration needs to be configured.